src/dataframe/dataframe.jl

```diff
@@ -1032,16 +1032,22 @@ hcat!(df1::DataFrame, df2::DataFrame;
     invoke(hcat!, Tuple{DataFrame, AbstractDataFrame}, df1, df2,
            makeunique=makeunique, copycols=copycols)::DataFrame
 
-hcat!(df::DataFrame, x::AbstractVector; makeunique::Bool=false, copycols::Bool=true) =
-    hcat!(df, DataFrame(AbstractVector[x], [:x1], copycols=copycols),
-          makeunique=makeunique, copycols=copycols)
-hcat!(x::AbstractVector, df::DataFrame; makeunique::Bool=false, copycols::Bool=true) =
-    hcat!(DataFrame(AbstractVector[x], [:x1], copycols=copycols), df,
-          makeunique=makeunique, copycols=copycols)
-hcat!(x, df::DataFrame; makeunique::Bool=false, copycols::Bool=true) =
-    throw(ArgumentError("x must be AbstractVector or AbstractDataFrame"))
 hcat!(df::DataFrame, x; makeunique::Bool=false, copycols::Bool=true) =
-    throw(ArgumentError("x must be AbstractVector or AbstractDataFrame"))
+    throw(ArgumentError("x must AbstractDataFrame"))
+hcat!(x, df::DataFrame; makeunique::Bool=false, copycols::Bool=true) =
+    throw(ArgumentError("x must AbstractDataFrame"))
```

Comment:
Maybe we should drop this method? Other packages may define methods for `x`, so the error may not be completely accurate (since other types than `AbstractDataFrame` may be supported).

Reply:
The issue is that other packages will not affect `hcat!` but only potentially `hcat`. Still - I understand we could drop `hcat(::Any, ::AbstractDataFrame)` right?

Reply:
Ah right. Yes, we could probably drop both methods.